rfc/kubernetes-ui-enhancements/kubernetes-ui-enhancements.md

```diff
@@ -0,0 +1,260 @@
+
+Initial rollout of this functionality would provide a terminology set tied to the Kubernetes provider but be generic enough to support extension to other cloud providers.
+
+Ideally, this approach will not degrade the UI experience for non-Kubernetes or mix provider setups. It will give users control on an application level and will improve the Kubernetes experience while remaining extensible.
+
+#### Concerns
+
+This approach may introduce a visual flicker during page load. Additionally, making UI terminology configurable at the application level is higher in complexity than an all or nothing approach such as a feature flag option.
+
+##### API
+
+Changes to terminology will be restricted to the UI. Where a mismatch would be introduced by updating a term in the UI, the API documentation will be updated to make note of the difference. Leaving endpoints unchanged will mitigate the risk of breaking current API consumers.
+
+##### Documentation
+
+Codelabs, API documentation, and user guides across Spinnaker.io will require updating. Terminology changes will need to be communicated to the community in advance through:
+
+- Terminology RFC
+- The Kubernetes SIG
+- Release Notes
+
+## Phase 2
+
+The scope of this project’s first phase has been detailed above. There exists a set of additional Kubernetes objects that may be helpful to expose to Spinnaker users. This section details a second phase of possible enhancements.
+
+## Custom CRDs
+
+Spinnaker currently offers [extension points](https://www.spinnaker.io/guides/developer/crd-extensions/) for advanced users seeking to accomplish a deeper integration with their own Custom Resource Definitions. This functionality offers users the ability to override the default model class for a CRD, implement a custom naming strategy, or offer a custom UI. To accomplish this, users must be comfortable working in Spinnaker’s code base and running a layered build. The complexity of this approach may act as a significant obstacle. Approaches such as general resource browser may present a more accessible alternative.
+
+## Config Map / Pod Relationship
+
+Phase 1 will expose Config Map objects. Phase 2 can surface the association with related Pods and detail what data, such as environment variables, is being used by a Pod.
+
+## Secrets
+
+Secrets are a volume type that can be used to store sensitive information. Due to the similarities they share with Config Maps, it should be possible to extend the Phase 1 UI to display them. Some existing tooling in the Kubernetes space elects not to show data field values. Before exposing these in Deck, it will be necessary to examine Spinnaker’s current authorization architecture [(fiat)](https://github.com/spinnaker/fiat) and design a secure means of rendering sensitive information directly in the UI.
```

Comment:
I think as a phase one, it may make sense to expose information about the secret (name), but not actual contents of the secret.  I think the part that is immediately relevant is knowledge that the secret exists and where it is.

Reply:
Interesting, I have seen other tooling follow this pattern. I believe this would resolve some (most?) of the security concerns but I think it would be helpful to dig in a bit further. I've added this to the [SIG agenda](https://docs.google.com/document/d/1db_yw1uru99Byvin4lQgm7aUZ9xMmvsARtEiiNUrmBo/edit#heading=h.rs4wkuo2ejno) so we can discuss further.

Reply:
Could include copy-to-clipboard links with generated `kubectl` commands to decode secrets for users that have `kubectl` and the required permissions.

Reply:
Update on the SIG discussion. The agenda notes are [here](https://docs.google.com/document/d/1db_yw1uru99Byvin4lQgm7aUZ9xMmvsARtEiiNUrmBo/edit#heading=h.rs4wkuo2ejno). The main takeway was that displaying secret metadata alongside the keys (not values) would be of use. I'm leaving this section unchanged just to manage the scope of the project but I think this points to a good path forward for displaying secrets.